File: 程式設計：資料結構答案檔/C++/102.cpp
#include <iostream>
#include <vector>

int main() {
    int m, n;
    std::cin >> m;
    std::cin >> n;

    std::vector<std::vector<int>> matrix(m, std::vector<int>(n));

    for (int i = 0; i < m; i++) {
        for (int j = 0; j < n; j++) {
            std::cin >> matrix[i][j];
        }
    }

    std::cout << "Original:\n";
    for (int i = 0; i < m; i++) {
        std::cout << "[";
        for (int j = 0; j < n; j++) {
            std::cout << matrix[i][j];
            if (j < n - 1) std::cout << " ";
        }
        std::cout << "]\n";
    }

    std::cout << "Transpose:\n";
    for (int i = 0; i < n; i++) {
        std::cout << "[";
        for (int j = 0; j < m; j++) {
            std::cout << matrix[j][i];
            if (j < m - 1) std::cout << " ";
        }
        std::cout << "]\n";
    }

    return 0;
}

File: 程式設計：資料結構答案檔/C#/102.cs
using System;

class Ans
{
    static void Main(string[] args)
    {
        int m = int.Parse(Console.ReadLine());
        int n = int.Parse(Console.ReadLine());

        int[,] matrix = new int[m, n];
        for (int i = 0; i < m; i++)
        {
            for (int j = 0; j < n; j++)
            {
                matrix[i, j] = int.Parse(Console.ReadLine());
            }
        }

        Console.WriteLine("Original:");
        for (int i = 0; i < m; i++)
        {
            Console.Write("[");
            for (int j = 0; j < n; j++)
            {
                Console.Write(matrix[i, j]);
                if (j < n - 1)
                {
                    Console.Write(" ");
                }
            }
            Console.WriteLine("]");
        }

        Console.WriteLine("Transpose:");
        for (int i = 0; i < n; i++)
        {
            Console.Write("[");
            for (int j = 0; j < m; j++)
            {
                Console.Write(matrix[j, i]);
                if (j < m - 1)
                {
                    Console.Write(" ");
                }
            }
            Console.WriteLine("]");
        }
    }
}
